iphone/Maps/Core/TextToSpeech/MWMTextToSpeech.mm

```diff
@@ -279,6 +282,46 @@ - (AVAudioPlayer *)audioPlayer {
   return _audioPlayer;
 }
 
+- (void)playRandomTestString {
+  if (![self isValid])
+    [self createVoice:[[self class] savedLanguage]];
+  
+  if (self.testStrings == nil)
+    return;
+  
+  [self speakOneString:self.testStrings[_testStringIndex]];
+  
+  if (++_testStringIndex >= self.testStrings.count)
+    _testStringIndex = 0;
+}
+
+- (NSArray *)testStrings {
```

Comment:
1. I think that would be better to move out all the logic from the 
```objc
- (NSArray *)testStrings {
...
    return _testStrings
}
```
to some builder method.

For now, by placing all this code into the getter the app will do this calculations for the every 
`self.testStrings` call (because the `- (NSArray *)testStrings` is actually overridden getter).
so in the `- (void)playRandomTestString` it will be done at least 3 time.
Lets make it once and than reuse the result value.

Or you can create a local variable like:
`NSArray * somename = [self testStrings];`
ant than use the `somename` inside the `playRandomTestString`.

The whole idea is to build the string only 1 time.

2. Should the `- (NSArray *)testStrings` be called multiple times at all? If not, maybe its better to build the strings once on the first access (here the dispatch_once will be useful).

Reply:
There's no reason to keep these strings in memory after settings are closed. Rebuilding them once on each initialization of the voice/nav dialog is okay.

Reply:
Oh I totally forgot about the caching of the strings. That was the whole point of the property in the first place. 
Anyway, I think I found a more elegant solution. I move the TTS test code to a new class called `TTSTester`. This class is initialized and removed with the voice settings view controller. The strings are loaded on the first press to the button and then cached. If the language changes, the strings are reloaded.